core/src/com/po/fuck/model/Room.java

```diff
@@ -33,11 +32,10 @@ private enum State {
     @SuppressWarnings("unchecked")
     protected Managed<InvisibleWall> walls[] = new Managed[4];
 
-    Room(Vector2 tillingPosition, float width, float height) {
+    public Room (Vector2 tillingPosition) {
         this.tillingPosition = tillingPosition;
-        this.geometryData = new GeometryData();
-        geometryData.setHeight(height);
-        geometryData.setWidth(width);
+        BasicSpriteInfo info = BasicSpriteManager.getBasicSpriteInfo(this.getClass());
+        this.geometryData = new GeometryData(new PositionData(), info.getSize());
```

Comment:
Can't we initialize position with `new Vector2(geometryData.getWidth() * tillingPosition.x, geometryData.getHeight() * tillingPosition.y)`, and change method `getPosition` accordingly ?